File: go-service/models/driver.go
package models

import "time"

// Driver represents a delivery driver
type Driver struct {
	ID                  int        `db:"Id" json:"id"`
	Name                string     `db:"Name" json:"name"`
	Phone               string     `db:"Phone" json:"phone"`
	LicenseNumber       *string    `db:"LicenseNumber" json:"licenseNumber,omitempty"`
	Vehicle             string     `db:"Vehicle" json:"vehicle"`
	VehicleRegistration *string    `db:"VehicleRegistration" json:"vehicleRegistration,omitempty"`
	Status              string     `db:"Status" json:"status"`
	Rating              float64    `db:"Rating" json:"rating"`
	IsActive            bool       `db:"IsActive" json:"isActive"`
	CreatedAt           time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt           *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

File: backend/Api/Models/Driver.cs
using System.Text.Json.Serialization;

namespace Api.Models
{
    public class Driver
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string? LicenseNumber { get; set; }
        public string Vehicle { get; set; } = string.Empty;
        public string? VehicleRegistration { get; set; }
        public string Status { get; set; } = "Available"; // Available, On Delivery, Off Duty
        public decimal Rating { get; set; } = 0;
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [JsonIgnore]
        public virtual ICollection<Delivery> Deliveries { get; set; } = new List<Delivery>();
    }
}
